test/integration/password_resets_test.rb

```diff
@@ -0,0 +1,69 @@
+require 'test_helper'
+class PasswordResetsTest < ActionDispatch::IntegrationTest
+  def setup 
+    ActionMailer::Base.deliveries.clear 
+    @user = users(:michael)
+  end
+
+  test "password resets" do
```

Comment:
there's a few issues with this test case:
  * non-descriptive test name
  * it's doing a lot, hard to follow & hard to debug when something breaks

please try to break each scenario (ex: each request + assertions) into its own test case with a descriptive title